doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1178 @@
+link_type           = "type"                    ; point-to-point or shared link type
+mst_boundary        = BIT                       ; enabled or disabled
+mst_boundary_proto  = BIT                       ; enabled or disabled
+```
+
+#### STP_INST_PORT_FLUSH_TABLE 
+
+```
+;Defines instance and port for which FDB Flush needs to be performed
+key                 = _STP_INST_PORT_FLUSH_TABLE:instane:ifname         ; FDB Flush instance id and port
+name           
+state               = "true"                                            
+```
+
+### Existing Tables
+Following already present APP_DB tables are also used for implementation of MSTP:
+
+#### STP_PORT_STATE_TABLE 
+The table holds the state of a port i.e forwarding, learning, blocking with respect to each instance.
+#### STP_VLAN_INSTANCE_TABLE 
+The table holds the VLAN to instance mapping.
+
+# SAI
+
+## Existing SAI Attributes
+Following table shows the existing SAI Attributes that will be used:
+
+|**Component**|**SAI Attribute**|
+| :- | :- |
+|STP Instance|SAI_STP_ATTR_VLAN_LIST
+||SAI_STP_ATTR_BRIDGE_ID|
+||SAI_STP_ATTR_PORT_LIST|
+|STP Port|SAI_STP_PORT_ATTR_STP |
+||SAI_STP_PORT_ATTR_BRIDGE_PORT|
+||SAI_STP_PORT_ATTR_STATE|
+|STP Port States|SAI_STP_PORT_STATE_LEARNING|
+||SAI_STP_PORT_STATE_FORWARDING|
+||SAI_STP_PORT_STATE_BLOCKING|
+|VLAN STP Instance|SAI_VLAN_ATTR_STP_INSTANCE|
+|Switch STP Attributes|SAI_SWITCH_ATTR_DEFAULT_STP_INST_ID|
+||SAI_SWITCH_ATTR_MAX_STP_INSTANCE|
+
+## New SAI Attributes
+MSTP design will not require any new SAI Attributes for Control Packet trap. We will use the existing traps as DMAC and LLC fields are identical for STP and MSTP
+
+# Additional Features
+
+## BPDU Guard
+BPDU guard feature will also be supported by MSTP. BPDU Guard feature disables the connected device ability to initiate or participate in STP on edge ports. When STP BPDUs are received on the port where BPDU guard is enabled the port will be shutdown. User can re-enable the port administratively after ensuring the BPDUs have stopped coming on the port.
+
+
+## Root Guard
+Root guard feature will also be supported by MSTP. 
+
+The Root Guard feature provides a way to enforce the root bridge placement in the network and allows STP to interoperate with user network bridges while still maintaining the bridged network topology that the administrator requires. When BPDUs are received on a root guard enabled port, the STP state will be moved to "Root inconsistent" state to indicate this condition. Once the port stops receiving superior BPDUs, Root Guard will automatically set the port back to a FORWARDING state.
+
+## Edge Port
+Edge ports immediately transition to the forwarding state upon activation and do not participate in STP topology calculations. 
+
+
+## Uplink Fast
+MSTP standard does not support uplink fast so uplink fast functionality will be disable for MSTP.
+
+
+#  Sequence Diagrams
+## MSTP global enable
+
```

Comment:
Please update the flow diagrams with the details, where the APP_DB changes/updates are done.

Reply:
Please update the diagram with APP_DB details.

Reply:
We will update the HLD diagram.

Reply:
updated